### PR TITLE
Pools V2 trade history API

### PR DESCRIPTION
**Pull request by Steven-Yulong-Yan**
Update trade history API

### CODE REVIEW

File: archetypes/Portfolio/History/index.tsx [hunk 1]
```diff
@@ -144,19 +144,19 @@ const CommitRow: React.FC<
...
-    tokenAddress,
+    tokenOutAddress,
     tokenDecimals,
-    collateralAmount,
-    collateralSymbol,
+    fee,
     transactionHashIn,
     transactionHashOut,
 }) => {
+    console.log(typeof price);
```
Comment (dweberdev) [inline archetypes/Portfolio/History/index.tsx:159]: Remove console.log

File: archetypes/Portfolio/History/index.tsx [hunk 2]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
+                                {tokenOutSymbol}
                             </div>
-                            &nbsp;
-                            <div className={type === 'LongMint' || type === 'LongBurn' ? 'green' : 'red'}>
-                                {type === 'LongMint' || type === 'LongBurn' ? 'Long' : 'Short'}
+                        </div>
+                    </TableRowCell>
+                    {/*Amount*/}
+                    <TableRowCell>
+                        {toApproxCurrency((price.toNumber() / 1e18) * (tokenInAmount.toNumber() / 1e18))}
```
Comment (ScaredIbis) [inline archetypes/Portfolio/History/index.tsx:243]: this should use the dynamic value for decimals (`tokenDecimals` on the trade history items)

File: archetypes/Portfolio/History/index.tsx [hunk 3]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
+                    </TableRowCell>
+                    {/*Tokens / Price*/}
+                    <TableRowCell>
+                        <div>{(tokenInAmount.toNumber() / 1e18).toFixed(2)} tokens</div>
+                        <div className="text-cool-gray-500">
+                            at {toApproxCurrency(price.toNumber() / 1e18)} USD/token
```
Comment (ScaredIbis) [inline archetypes/Portfolio/History/index.tsx:249]: I have updated the api to return name and symbol of the pricing token, we'll need to replace any hardcoded `USD`'s with this

File: archetypes/Portfolio/History/index.tsx [hunk 4]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
+                        </div>
+                    </TableRowCell>
+                    {/*Protocol Fee*/}
+                    <TableRowCell>{toApproxCurrency(fee.toNumber() / 1e18)} USDC</TableRowCell>
```
Comment (ScaredIbis) [inline archetypes/Portfolio/History/index.tsx:253]: use dynamic decimals here

File: libs/utils/tradeHistoryAPI.ts [hunk 5]
```diff
@@ -2,41 +2,54 @@ import { ARBITRUM, ARBITRUM_RINKEBY } from '@libs/constants';
 import BigNumber from 'bignumber.js';
 
 // Base API URL
-const BASE_TRADE_HISTORY_API = process.env.NEXT_PUBLIC_TRADE_HISTORY_API;
+// const BASE_TRADE_HISTORY_API = process.env.NEXT_PUBLIC_TRADE_HISTORY_API;
+const BASE_TRADE_HISTORY_API = 'https://dev.api.tracer.finance/poolsv2/tradeHistory';
```
Comment (ScaredIbis) [inline libs/utils/tradeHistoryAPI.ts:6]: assume this was a temp change, need to undo it

File: libs/utils/tradeHistoryAPI.ts [hunk 6]
```diff
@@ -2,41 +2,54 @@ import { ARBITRUM, ARBITRUM_RINKEBY } from '@libs/constants';
...
+    tokenInAddress: string;
+    tokenInSymbol: string;
+    tokenInName: string;
+    tokenInAmount: string;
+    price: string;
+    fee: string;
+    tokenOutAddress: string;
+    tokenOutSymbol: string;
+    tokenOutName: string;
+    tokenOutAmount: string;
```
Comment (ScaredIbis) [inline libs/utils/tradeHistoryAPI.ts:33]: this type has been updated to include the following fields:

`priceTokenAddress`: string
`priceTokenName`: string
`priceTokenSymbol`: string

**Review by ScaredIbis — CHANGES_REQUESTED** [review]

File: archetypes/Portfolio/History/index.tsx [hunk 7]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
-                        {tokenSymbol}
-                    </div>
-                </div>
-            </TableRowCell>
-            {type === 'LongBurn' || type === 'ShortBurn' ? (
-                <>
+                    </TableRowCell>
                     <TableRowCell>
-                        <div>{tokenAmount.toFixed(2)} tokens</div>
+                        <div>{(tokenInAmount.toNumber() / 1e18).toFixed(2)} tokens</div>
```
Comment (dospore) [inline archetypes/Portfolio/History/index.tsx:314]: Dynamic decimals here as well

File: archetypes/Portfolio/History/index.tsx [hunk 8]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
                         <div className="text-cool-gray-500">
-                            at {toApproxCurrency(tokenPrice)} {collateralSymbol}/token
+                            at {toApproxCurrency(price.toNumber() / 1e18)} USD/token
+                        </div>
+                    </TableRowCell>
+                    <TableRowCell>
+                        {toApproxCurrency((price.toNumber() / 1e18) * (tokenInAmount.toNumber() / 1e18))}
```
Comment (dospore) [inline archetypes/Portfolio/History/index.tsx:320]: and here

File: archetypes/Portfolio/History/index.tsx [hunk 9]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
-            </TableRowCell>
-            {type === 'LongBurn' || type === 'ShortBurn' ? (
-                <>
+                    </TableRowCell>
                     <TableRowCell>
-                        <div>{tokenAmount.toFixed(2)} tokens</div>
+                        <div>{(tokenInAmount.toNumber() / 1e18).toFixed(2)} tokens</div>
                         <div className="text-cool-gray-500">
-                            at {toApproxCurrency(tokenPrice)} {collateralSymbol}/token
+                            at {toApproxCurrency(price.toNumber() / 1e18)} USD/token
```
Comment (dospore) [inline archetypes/Portfolio/History/index.tsx:316]: dynamic decimals

File: archetypes/Portfolio/History/index.tsx [hunk 10]
```diff
@@ -162,82 +205,229 @@ const CommitRow: React.FC<
...
+                        </div>
+                    </TableRowCell>
+                    <TableRowCell>
+                        {toApproxCurrency((price.toNumber() / 1e18) * (tokenInAmount.toNumber() / 1e18))}
+                    </TableRowCell>
+                    <TableRowCell>{toApproxCurrency(fee.toNumber() / 1e18)} USDC</TableRowCell>
```
Comment (dospore) [inline archetypes/Portfolio/History/index.tsx:322]: I dont think this will always be USDC so will have to use `quoteToken.symbol` here as well

**Review by dospore — APPROVED** [review]

**Review by ScaredIbis — APPROVED** [review]

**Pull request merged**